ui/lib/kv/addon/routes/secret.js

```diff
@@ -12,16 +12,21 @@ export default class KvSecretRoute extends Route {
   @service secretMountPath;
 
   async fetchSecretData(backend, path) {
-    return await this.store.queryRecord('kv/data', { backend, path }).catch((error) => error.httpStatus);
+    return await this.store.queryRecord('kv/data', { backend, path }).catch(() => {
+      // return empty record to access capability getters on model
+      return this.store.createRecord('kv/data', { backend, path });
```

Comment:
It'd probably make more sense to have some sort of `kv-permissions` model but that may be too much of a lift right now. Just something to consider for later..